cohttp-lwt-unix/src/input_channel.ml

```diff
@@ -0,0 +1,24 @@
+open Lwt.Infix
+
+type t = { buf : Bytebuffer.t; chan : Lwt_io.input_channel }
```

Comment:
Adding a buffer layer alongside `Lwt_io` as it helps in driving the core IO layer that needs access to the internal view within the buffer to feed data to the new parser.

Reply:
Before this is considered ready, we might want a way to grow the buffer too I suppose (with an upper bound beyond which its considered an error maybe?)

Reply:
Could you expand on this a bit more. When does the buffer need to grow?

Reply:
I don't think it needs to grow as such as long we pick a value that we feel is reasonable to parse the request + headers. But I noticed that the body decoding implementation in cohttp references that it'd like to read up to a max of 32k bytes at a time when reading chunked bodies https://github.com/mirage/ocaml-cohttp/blob/efe05994fcab28c321caae5fb3ece599738979ce/cohttp/src/transfer_io.ml#L52 So I figured if we want to start with a smaller buffer than that, we'll need to grow it if we still want to eventually read 32k bytes when we get to bodies. Or we could just pick a large enough buffer to start with and never touch it again.

Reply:
32K sounds like a reasonable maximum. I suppose we should determine the default buffer size experimentally and let the user customize it if need be.

Reply:
For reference, I went back through the history to find the PR that first introduced the limit: https://github.com/mirage/ocaml-cohttp/pull/330 Looks like at that stage it was arbitrary. I think it would be nice to make it configurable, something similar to `Lwt_io.set_default_buffer_size io_buf`